### PR TITLE
Adjust `DispatcherSpec`

### LINKED ISSUE #3029: Nondeterministic failures in `DispatcherSpec`
Seems to happen much more often in CI. I have yet to reproduce locally. Example output:

```
[error] sbt.ForkMain$ForkError: java.lang.NullPointerException: null
[error]   at org.specs2.reporter.TextPrinter.printSkipped(TextPrinter.scala:181)
[error]   at org.specs2.reporter.TextPrinter.printSkipped$(TextPrinter.scala:178)
[error]   at org.specs2.reporter.TextPrinter$.printSkipped(TextPrinter.scala:292)
[error]   at org.specs2.reporter.TextPrinter.printResult$1(TextPrinter.scala:130)
[error]   at org.specs2.reporter.TextPrinter.printExecutable(TextPrinter.scala:146)
[error]   at org.specs2.reporter.TextPrinter.printExecutable$(TextPrinter.scala:118)
[error]   at org.specs2.reporter.TextPrinter$.printExecutable(TextPrinter.scala:292)
[error]   at org.specs2.reporter.TextPrinter.$anonfun$printFragment$2(TextPrinter.scala:100)
[error]   at org.specs2.control.eff.Arrs.$anonfun$map$2(Eff.scala:368)
[error]   at org.specs2.control.eff.Arrs.go$1(Eff.scala:383)
[error]   at org.specs2.control.eff.Arrs.apply(Eff.scala:399)
[error]   at org.specs2.control.eff.Interpret.$anonfun$interpretLoop$11(Interpret.scala:198)
[error]   at org.specs2.control.eff.Arrs.go$1(Eff.scala:380)
[error]   at org.specs2.control.eff.Arrs.apply(Eff.scala:399)
[error]   at org.specs2.control.eff.CollectedUnions.$anonfun$othersEff$1(Unions.scala:97)
[error]   at org.specs2.control.eff.Arrs.go$1(Eff.scala:383)
[error]   at org.specs2.control.eff.Arrs.apply(Eff.scala:399)
[error]   at org.specs2.control.eff.Interpret.$anonfun$interpretLoop$11(Interpret.scala:198)
[error]   at org.specs2.control.eff.Arrs.go$1(Eff.scala:380)
[error]   at org.specs2.control.eff.Arrs.apply(Eff.scala:399)
[error]   at org.specs2.control.eff.CollectedUnions.$anonfun$othersEff$1(Unions.scala:97)
[error]   at org.specs2.control.eff.Arrs.go$1(Eff.scala:380)
[error]   at org.specs2.control.eff.Arrs.apply(Eff.scala:399)
[error]   at org.specs2.control.eff.Interpret.$anonfun$interpretLoop$11(Interpret.scala:198)
[error]   at org.specs2.control.eff.Arrs.go$1(Eff.scala:380)
[error]   at org.specs2.control.eff.Arrs.apply(Eff.scala:399)
[error]   at org.specs2.control.eff.CollectedUnions.$anonfun$othersEff$1(Unions.scala:97)
[error]   at org.specs2.control.eff.Arrs.go$1(Eff.scala:383)
[error]   at org.specs2.control.eff.Arrs.apply(Eff.scala:399)
[error]   at org.specs2.control.eff.Interpret.$anonfun$interpretLoop$11(Interpret.scala:198)
[error]   at org.specs2.control.eff.Arrs.go$1(Eff.scala:380)
[error]   at org.specs2.control.eff.Arrs.apply(Eff.scala:399)
[error]   at org.specs2.control.eff.CollectedUnions.$anonfun$othersEff$1(Unions.scala:97)
[error]   at org.specs2.control.eff.Arrs.go$1(Eff.scala:383)
[error]   at org.specs2.control.eff.Arrs.apply(Eff.scala:399)
[error]   at org.specs2.control.eff.IntoPolyLower4$$anon$15.$anonfun$apply$62(IntoPoly.scala:220)
[error]   at org.specs2.control.eff.Arrs.go$1(Eff.scala:380)
[error]   at org.specs2.control.eff.Arrs.apply(Eff.scala:399)
[error]   at org.specs2.control.eff.IntoPolyLower4$$anon$15.$anonfun$apply$62(IntoPoly.scala:220)
[error]   at org.specs2.control.eff.Arrs.go$1(Eff.scala:380)
[error]   at org.specs2.control.eff.Arrs.apply(Eff.scala:399)
[error]   at org.specs2.control.eff.EffInterpretation.$anonfun$detachA$8(Eff.scala:316)
[error]   at org.specs2.fp.Monad.$anonfun$map$2(Monad.scala:22)
[error]   at org.specs2.control.eff.TimedFuture$$anon$2.$anonfun$point$2(FutureEffect.scala:57)
[error]   at org.specs2.control.eff.TimedFuture.$anonfun$runNow$1(FutureEffect.scala:15)
[error]   at scala.Option.fold(Option.scala:263)
[error]   at org.specs2.control.eff.TimedFuture.runNow(FutureEffect.scala:16)
[error]   at org.specs2.control.eff.TimedFuture$$anon$2.$anonfun$bind$2(FutureEffect.scala:60)
[error]   at scala.concurrent.impl.Promise$Transformation.run(Promise.scala:470)
[error]   at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
[error]   at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
[error]   at java.lang.Thread.run(Thread.java:750)
```

No indication of which test is failing. The NPE itself seems to indicate a `Skipped` with a `null` message, but I don't see how such a thing could come to exist.

### PR DESCRIPTION
**Pull request by djspiewak**
Hopefully fixes #3029

### CODE REVIEW

**Comment by djspiewak** [comment]
This doesn't resolve it :-(

File: tests/shared/src/test/scala/cats/effect/std/DispatcherSpec.scala [hunk 1]
```diff
@@ -25,32 +25,34 @@ import scala.concurrent.duration._
 
 class DispatcherSpec extends BaseSpec {
 
-  "sequential dispatcher (await = true)" should {
-    val D = Dispatcher.sequential[IO](await = true)
+  override def executionTimeout = 30.seconds
```
Comment (armanbilge) [inline tests/shared/src/test/scala/cats/effect/std/DispatcherSpec.scala:28]: Is this the fix?

**Review by armanbilge — APPROVED** [review]
👍 if it works!

File: tests/shared/src/test/scala/cats/effect/std/DispatcherSpec.scala (hunk 1 above)
Comment (djspiewak) [reply tests/shared/src/test/scala/cats/effect/std/DispatcherSpec.scala:28]: I think so! :-P

**Pull request merged**